fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/DerivationsProperty.java

```diff
@@ -336,6 +338,15 @@ public Derivations visitInsertPlan(@Nonnull final RecordQueryInsertPlan insertPl
             return new Derivations(resultValuesBuilder.build(), localValuesBuilder.build());
         }
 
+        @Nonnull
+        @Override
+        public Derivations visitTableFunctionPlan(@Nonnull final RecordQueryTableFunctionPlan tableFunctionPlan) {
+            final var streamingValue = tableFunctionPlan.getValue();
+            final var elementType = streamingValue.getResultType();
+            final var values = ImmutableList.<Value>of(new FirstOrDefaultStreamingValue(streamingValue, new ThrowsValue(elementType)));
```

Comment:
Since these values are fake anyway, when we spoke I thought we had agreed on using one of these for both arrays and the new stream value.

Reply:
Yes, but this can only happen once `StreamingValue` is able to handle both arrays (explode) and streamed values (e.g. range). Then we can use `FirstOrDefaultStreamingValue` that expects an underlying `StreamingValue`. please let me know if this is aligned with your thinking.

By the way, I think there is a different in the empty semantics between `FirstOrDefaultStreamingValue` and `FirstOrDefaultValue`. We can talk about it offline if you want.